src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -600,6 +600,23 @@ private static string GetSharedModulePath()
 #endif
         }
 
+#if !UNIX
+        /// <summary>
+        /// Get the path to the Windows PowerShell module directory under the
+        /// System32 directory on Windows (the Windows PowerShell $PSHOME).
+        /// </summary>
+        /// <returns>The path of the Windows PowerShell system module directory.</returns>
+        internal static string GetWindowsPowerShellPSHomeModulePath()
+        {
+            if (!String.IsNullOrEmpty(InternalTestHooks.TestWindowsPowerShellPSHomeLocation))
+            {
+                return InternalTestHooks.TestWindowsPowerShellPSHomeLocation;
+            }
+
+            return Path.Combine(System.Environment.SystemDirectory, "WindowsPowerShell", "v1.0", "Modules");
```

Comment:
It's better to have a static field keeping the value, so we only need to calculate it once.

Reply:
That's what I wanted to do. I assume the environment variable can't change then?

Reply:
`System.Environment.SystemDirectory` doesn't use the environment variable. It calls the win32 API `GetSystemDirectoryW`. See https://msdn.microsoft.com/en-us/library/windows/desktop/ms724373(v=vs.85).aspx